### PR TITLE
Add pre-commit + eslint with Vue support

### LINKED ISSUE #660: Add eslint + pre-commit to lint frontend source code
eslint template used by MozillaSecurity:

```json
{
  "name": "ts-project-template",
  "version": "1.0.0",
  "description": "Typescript Project Template",
  "keywords": [
    "template"
  ],
  "homepage": "https://github.com/MozillaSecurity/ts-project-template",
  "bugs": {
    "url": "https://github.com/MozillaSecurity/ts-project-template/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+ssh://git@github.com/MozillaSecurity/ts-project-template.git"
  },
  "license": "MPL-2.0",
  "author": "Jason Kratzer <jkratzer@mozilla.com>",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "tsc",
    "build:dev": "tsc --sourceMap",
    "prepublishOnly": "npm run build",
    "lint": "eslint ./src/ --ext ts",
    "lint:fix": "eslint ./src/ --ext ts --fix",
    "test": "jest --silent"
  },
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged",
      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
    }
  },
  "commitlint": {
    "extends": [
      "@commitlint/config-conventional"
    ]
  },
  "lint-staged": {
    "**.ts": [
      "npx prettier --write",
      "eslint --cache ./src/ --fix"
    ],
    "package.json": "sort-package-json"
  },
  "prettier": {
    "jsdocSingleLineComment": false,
    "printWidth": 100,
    "semi": false,
    "tsdoc": true
  },
  "jest": {
    "preset": "ts-jest",
    "testEnvironment": "node"
  },
  "dependencies": {
    "typescript": "^4.2.3"
  },
  "devDependencies": {
    "@commitlint/cli": "^12.0.1",
    "@commitlint/config-conventional": "^12.0.1",
    "@semantic-release/changelog": "^5.0.1",
    "@semantic-release/git": "^9.0.0",
    "@semantic-release/npm": "^7.0.10",
    "@types/jest": "^26.0.21",
    "@typescript-eslint/eslint-plugin": "^4.19.0",
    "@typescript-eslint/parser": "^4.19.0",
    "eslint": "^7.22.0",
    "eslint-config-prettier": "^8.1.0",
    "eslint-plugin-jest": "^24.3.2",
    "eslint-plugin-jsdoc": "^32.3.0",
    "eslint-plugin-prettier": "^3.3.1",
    "eslint-plugin-tsdoc": "^0.2.11",
    "husky": "^4.3.8",
    "jest": "^26.6.3",
    "lint-staged": "^10.5.4",
    "prettier": "^2.2.1",
    "prettier-plugin-jsdoc": "^0.3.13",
    "semantic-release": "^17.4.2",
    "sort-package-json": "^1.49.0",
    "ts-jest": "^26.5.4",
    "ts-node": "^9.1.1"
  },
  "release": {
    "branches": [
      "main"
    ],
    "plugins": [
      "@semantic-release/commit-analyzer",
      "@semantic-release/release-notes-generator",
      "@semantic-release/changelog",
      "@semantic-release/npm",
      [
        "@semantic-release/git",
        {
          "assets": [
            "CHANGELOG.md",
            "package.json"
          ],
          "message": "chore(release): ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}"
        }
      ]
    ]
  }
}
```

### PR DESCRIPTION
**Pull request by La0**
Closes #660

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+---
+extends:
+  - eslint:recommended
+  - plugin:vue/strongly-recommended
+  - standard
```
Comment (pyoor) [inline .eslintrc.yml:5]: We use prettier in all other JS projects.  Can you swap this with prettier?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    hooks:
+    -   id: eslint
+        language: node
+        args: ['--fix', '--max-warnings', '0']
+        files: ^server/frontend/.*\.(js|vue)$
+        types: [file]
+        additional_dependencies:
+        - eslint@7.25.0
+        - eslint-plugin-vue@7.9.0
+        - eslint-config-standard@16.0.2
```
Comment (pyoor) [inline .pre-commit-config.yaml:13]: Same here `s/standard/prettier/`.

**Review by pyoor — APPROVED** [review]
Minor nit regarding standard vs. prettier.  Otherwise, looks good.

**Comment by La0** [comment]
I replaced standard by prettier: generated fixes are updated and I just had to make some minor fixes on one file

**Review by EvaBardou — APPROVED** [review]
Looks good on my side

**Pull request merged**